### PR TITLE
Replace UNION ALL with java logic.

### PR DESCRIPTION
**Pull request by agorina**
This is based on the latest load testing findings. The goal is to reduce the number of SQL query variations.

### CODE REVIEW

**Comment by agorina** [comment]
Merging, will address comments if needed.

**Comment by jtreuting** [comment]
the idea is that if there are 3 queries they will have the same compiled query under the hood and be much faster?

**Comment by agorina** [comment]
Yup, that is the idea.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/ReportQueryProvider.java [hunk 1]
```diff
@@ -4,14 +4,14 @@
 
 
 /**
- * Implementation of this interface is responsible for converting a {@link ReportQuery} to a sql query string
+ * Implementation of this interface is responsible for converting a {@link ReportQuery} to a sql query strings
  */
 public interface ReportQueryProvider {
     /**
      * Converts a given {@link ReportQuery} to a sql query string
```
Comment (malaffoon) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/ReportQueryProvider.java]: Perhaps some explanation of what the returned collection of sql is?

Comment (agorina) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/ReportQueryProvider.java]: Will do.

**Pull request merged**